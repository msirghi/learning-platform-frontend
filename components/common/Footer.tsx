import React from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles, Theme } from '@material-ui/core/styles';
import Link from '@material-ui/core/Link';
import MailIcon from '@material-ui/icons/Mail';
import HomeIcon from '@material-ui/icons/Home';
import styles from '../../styles/modules/Shared.module.scss';
import { useTranslation } from '../../i18n';

const Copyright: React.FC<{ siteName: string }> = ({ siteName }) => {
  return (
    <Typography variant='body2'>
      {'Copyright Â© '}
      <Link color='inherit' href='https://material-ui.com/'>
        {siteName}
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
};

const useStyles = makeStyles<Theme, { drawerOpened: boolean }>((theme) => ({
  footer: {
    padding: theme.spacing(4, 3),
    marginTop: 'auto',
    backgroundColor: '#202020',
    color: '#fff'
  },
  wrapper: ({ drawerOpened }) => ({
    '@media (min-width: 400px)': {
      marginLeft: 0
    },
    '@media (min-width: 600px)': {
      marginLeft: drawerOpened ? '35%' : '20%'
    },
    '@media (min-width: 1000px)': {
      marginLeft: drawerOpened ? '35%' : '15%'
    }
  })
}));

type Props = {
  drawerOpened: boolean;
};

export const SiteFooter: React.FC<Props> = ({ drawerOpened }) => {
  const classes = useStyles({ drawerOpened });
  const { t } = useTranslation();

  return (
    <footer className={classes.footer}>
      <div className={`${styles.footerWrapper} ${classes.wrapper}`}>
        <div className={styles.footerAboutUs}>
          <div>{t('common:footer.aboutUs')}</div>
          <div className={styles.footerUsefulLinksEvenLink}>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio odit numquam, omnis
            adipisci ex aperiam ipsum consectetur assumenda cupiditate debitis? Lorem ipsum dolor
            sit amet consectetur adipisicing elit. Optio, nemo.
          </div>
        </div>
        <div className={styles.footerUsefulLinks}>
          <div>{t('common:footer.usefulLinks')}</div>
          <div className={styles.footerUsefulLinksContent}>
            <div>
              <Link color='inherit' href='https://material-ui.com/'>
                Link 1
              </Link>
              <div className={styles.footerUsefulLinksEvenLink}>
                <Link color='inherit' href='https://material-ui.com/'>
                  Link 2
                </Link>
              </div>
            </div>
            <div className={styles.footerUsefulLinksSecondCol}>
              <Link color='inherit' href='https://material-ui.com/'>
                Link 4
              </Link>
              <div className={styles.footerUsefulLinksEvenLink}>
                <Link color='inherit' href='https://material-ui.com/'>
                  Link 3
                </Link>
              </div>
            </div>
          </div>
        </div>

        <div>
          <div>{t('common:footer.contacts')}</div>
          <div className={styles.footerIconText}>
            <MailIcon /> <span>support@site.com</span>
          </div>
          <div className={styles.footerIconText}>
            <HomeIcon /> <span>Chisinau, str.Cilene Bune 69</span>
          </div>
        </div>
        <div className={styles.footerCopyright}>
          <Copyright siteName={t('common:siteName')} />
        </div>
      </div>
    </footer>
  );
};
