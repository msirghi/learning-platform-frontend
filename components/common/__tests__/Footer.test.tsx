import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { SiteFooter } from '../Footer';

describe('Footer component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<SiteFooter drawerOpened />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
