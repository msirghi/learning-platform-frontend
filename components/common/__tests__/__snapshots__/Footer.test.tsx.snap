// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should match the snapshot 1`] = `
<footer
  className="makeStyles-footer-1"
>
  <div
    className="footerWrapper makeStyles-wrapper-2 makeStyles-wrapper-3"
  >
    <div
      className="footerAboutUs"
    >
      <div>
        common:footer.aboutUs
      </div>
      <div>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio odit numquam, omnis adipisci ex aperiam ipsum consectetur assumenda cupiditate debitis? Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio, nemo.
      </div>
    </div>
    <div
      className="footerUsefulLinks"
    >
      <div>
        common:footer.usefulLinks
      </div>
      <div
        className="footerUsefulLinksContent"
      >
        <div>
          <WithStyles(ForwardRef(Link))
            color="inherit"
            href="https://material-ui.com/"
          >
            Link 1
          </WithStyles(ForwardRef(Link))>
          <div
            className="footerUsefulLinksEvenLink"
          >
            <WithStyles(ForwardRef(Link))
              color="inherit"
              href="https://material-ui.com/"
            >
              Link 2
            </WithStyles(ForwardRef(Link))>
          </div>
        </div>
        <div
          className="footerUsefulLinksSecondCol"
        >
          <WithStyles(ForwardRef(Link))
            color="inherit"
            href="https://material-ui.com/"
          >
            Link 4
          </WithStyles(ForwardRef(Link))>
          <div
            className="footerUsefulLinksEvenLink"
          >
            <WithStyles(ForwardRef(Link))
              color="inherit"
              href="https://material-ui.com/"
            >
              Link 3
            </WithStyles(ForwardRef(Link))>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        common:footer.contacts
      </div>
      <div
        className="footerIconText"
      >
        <Memo([object Object]) />
         
        <span>
          support@site.com
        </span>
      </div>
      <div
        className="footerIconText"
      >
        <Memo([object Object]) />
         
        <span>
          Chisinau, str.Cilene Bune 69
        </span>
      </div>
    </div>
    <div
      className="footerCopyright"
    >
      <Copyright
        siteName="common:siteName"
      />
    </div>
  </div>
</footer>
`;
